src/gov/nist/javax/sip/stack/IOHandler.java

```diff
@@ -106,9 +106,9 @@ protected Socket getSocket(String key) {
 
     protected void removeSocket(String key) {
         socketTable.remove(key);
-        if ( socketCreationMap.get(key) != null ) {
```

Comment:
This kinf of preliminar checking use to be done to enhance performance. After initial light checking, it needs to recheck in proper critical section, which its what is missing here, hence NPE